src/logic/getWordsFromGrid.js

```diff
@@ -0,0 +1,46 @@
+function getHorizontalWordsFromGrid(grid) {
+  let words = [];
+
+  for (const row of grid) {
+    let word = "";
+    for (const letter of row) {
+      if (letter != "") {
+        word += letter;
+      } else {
+        if (word.length > 1) {
+          words.push(word);
+        }
+        word = "";
+      }
+    }
+    // Also push the word if we reach the end of the row
+    if (word.length > 1) {
+      words.push(word);
+    }
+  }
+
+  return words;
+}
+
+export function getWordsFromGrid(grid) {
+  // The simple method of grid transposition used in this function
+  // relies on an equal number of rows and columns,
+  // so throw errors if that isn't the case
+  if (grid.length != grid[0]?.length) {
+    throw new Error(
+      `The number of columns and number of rows in the grid must be equal.`
+    );
+  }
+
+  const numColumnsPerRow = grid.map((row) => row.length);
+  if (Math.min(...numColumnsPerRow) != Math.max(...numColumnsPerRow)) {
+    throw new Error(`All of the rows in the grid must have the same length.`);
+  }
+
+  const transposedGrid = grid.map((_, index) => grid.map((row) => row[index]));
```

Comment:
It would be nice to have this idiom in a function somewhere, so that it's just `transpose(grid)` here.

Reply:
Good idea. I use this in a few different places across multiple games, so I'll open a separate PR to consolidate it into the the skedwards88/word_logic package.

Reply:
> so I'll open a separate PR 

(For future ref: https://github.com/skedwards88/word_logic/pull/4)